### ISSUE TITLE
Steam Link, Sound won't send to TV, plays through speakers on computer

### ISSUE THREAD

**Issue by BloodyIron** [ValveSoftware/steam-for-linux#5793]
Same specs as : https://github.com/ValveSoftware/steam-for-linux/issues/5791

I've been trying really hard to have a good experience with my Steam Link and Steam Gamepad interfacing with my computer, but I've seriously not had a fun time.

This ticket is about this issue:
* Sound only plays on the computer, it won't send to the Steam Link. Yes, I've gone through troubleshooting, yes the Steam Link is set to stream audio. Yes I get audio in the main Steam Link menu, but not in Steam Big Picture steamed from my Linux gaming rig.

**Comment by kisak-valve** [comment]
Hello @BloodyIron, can you confirm that pulseaudio is running on the host system?

**Comment by BloodyIron** [comment]
@kisak-valve yeah pulse is being used, I know that for sure. :)

**Comment by bryanmr** [comment]
I have the same issue when I turn on simultaneous output in Pulse Audio on Ubuntu 18.04. If I have it disabled, everything seems to work fine. I get audio in the menu. I also get audio when I start streaming after a game is already launched.

I can confirm it is creating the Monitor in PulseAudio in both cases, but what it monitors can vary.

![screenshot from 2018-11-30 14-34-16](https://user-images.githubusercontent.com/18637958/49311835-44c54300-f4b0-11e8-81a0-5fdcf322281b.png)

@BloodyIron Is there something non-standard with your configuration? An equalizer or something else that isn't a stock configuration? I'll post back here if I discover a consistent understanding of what is happening or finally figure out how PulseAudio works.

**Comment by BloodyIron** [comment]
@bryanmr I'm using stock pulse audio, I don't really recall any big changes to it.

**Comment by queequegaz** [comment]
@bryanmr I'm having this exact issue (but it just started a week or two ago).  I don't know what you mean about turning on "simultaneous output in Pulse Audio", but it sounds like you have a workaround?  My PulseAudio "Recording" tab is completely empty (Steam is not listed as an application under Recording even once, let along twice), unlike yours that shows a couple Steam entries.  Do you have a workaround, and if so can you walk me through how to do it?

**Comment by bryanmr** [comment]
@queequegaz I'm assuming that we have a different issue that has the same root cause. It seems likely that the way Steam is creating the monitor isn't having a consistent result. Do you have sound while in the Big Picture menu?

Are you saying when you look at pavucontrol there is nothing in the Recording tab while you are trying to stream? Does the result change if you connect when the game has already been launched?

I can workaround this (it appears) by changing my output device to a standard device instead of the simultaneous one. (I set this option via paprefs) I'd start by seeing if changing the output device or connecting while already streaming works around the issue for you like it does for me. If you hear sound while in Big Picture, the Recording monitor most likely is getting created correctly to begin with.

**Comment by queequegaz** [comment]
@bryanmr I do NOT have sound on the streaming client while in the Big Picture menu.  Nor do I get sound if I connect to a game that has already launched.  

You're correct in that the "Recording" tab DOES show a single entry once streaming starts.  It is the same entry that shows up whether connecting before or after the game has launched.

Sounds like my issue is the same as @BloodyIron .  Thank you for responding.

I should note that I have several client devices (multiple Steam Links, a Windows client, and a Linux Client) and the sound suddenly stopped streaming to all of them (so it's definitely a host issue).

**Comment by queequegaz** [comment]
@bryanmr Sorry to bother you, but in the screencap you posted earlier there were two separate entries under the "Recording" tab.  When I'm streaming I only get a single entry (the 2nd one marked "Capture" in your screencap does not show up).  Do you always have both these entries show up when streaming?  @BloodyIron , do you have two entries or one when you aren't hearing audio through the stream?

**Comment by bryanmr** [comment]
@queequegaz I only get the one device when it works. Sometimes when it fails I only get one device as well. I'd paste the output from pacmd to show how things are, but I think the screenshots are easier. I'm also including gnome-control-center below.

I can work around this by making ~/.steam/ubuntu12_32/steam_monitor into a wrapper that changes my PulseAudio setting, it appears. It takes two arguments that I can just pass through. I'm going to open a separate issue once I have a complete handle on what is going on and what works and not.

Failing with one: 
![screenshot from 2018-12-13 10-25-25](https://user-images.githubusercontent.com/18637958/49948570-b7c3b600-fec1-11e8-86b7-e56f9b7aa1cc.png)

Success consistently: 
![screenshot from 2018-12-13 10-29-52](https://user-images.githubusercontent.com/18637958/49948794-1f7a0100-fec2-11e8-8cb0-494a86ec9106.png)

My weird settings (paprefs for simultaneous output): 
![screenshot from 2018-12-13 10-16-36](https://user-images.githubusercontent.com/18637958/49948171-e7be8980-fec0-11e8-9d94-1d2e2da21ca0.png)

**Comment by WizardStan** [comment]
Oh wow! I'd been fighting this issue for hours, flipping settings and reinstalling drivers and whatever else everyone suggested short of re-installing my OS.
As you say, the new input device was muted.  I needed to flip "show" to be "all input devices" so it would show me the muted monitors, but once I did that, unmuted, no problems.

**Comment by mendhak** [comment]
Future searchers, WizardStan's comment helped me. I was also unable to get audio working on Ubuntu going to Steam Link.  

First I installed Pulse Audio volume control

    sudo apt install pavucontrol

Then in this UI, I went to the Input Devices tab, changed the bottom dropdown to 'all input devices' and then unmuted the one that appeared muted.  Audio started working right away in Steam Link.  Screenshot to help: 

![image](https://user-images.githubusercontent.com/746276/198902134-e1f06c5b-9e75-43c3-8c55-66b6da3d6388.png)
